fighters/dedede/src/opff.rs

```diff
@@ -129,12 +136,124 @@ unsafe fn gordo_recatch(boma: &mut BattleObjectModuleAccessor, frame: f32, fight
+    //Removing shaking from ftilt, dair, up air, rapid jab with the mask on
+    if fighter.is_motion_one_of(&[Hash40::new("attack_s3_s"), Hash40::new("attack_air_hi"), Hash40::new("attack_100")])
+    && mask_is_exist{
+        ShakeModule::stop(boma);
+    }
+
+    //ensuring the flag resets when the mask does not exist
+    if mask_is_equipped && !mask_is_exist{
+        VarModule::off_flag(boma.object(), vars::dedede::instance::EQUIP_MASK);
+    }
+
+    // remove mask on entry and death
+    if fighter.is_status_one_of(&[*FIGHTER_STATUS_KIND_ENTRY, *FIGHTER_STATUS_KIND_DEAD])
+    && mask_is_equipped {
+        ArticleModule::remove_exist(boma, *FIGHTER_DEDEDE_GENERATE_ARTICLE_MASK, app::ArticleOperationTarget(*ARTICLE_OPE_TARGET_ALL));
+    }
+}
+
 unsafe fn super_jump_fail_edge_cancel(fighter: &mut L2CFighterCommon){
     if fighter.is_status(*FIGHTER_DEDEDE_STATUS_KIND_SPECIAL_HI_FAILURE) && fighter.global_table[SITUATION_KIND] == SITUATION_KIND_AIR {
         StatusModule::change_status_force(fighter.boma(), *FIGHTER_STATUS_KIND_FALL, false);
     }
 }
 
+unsafe fn jet_hammer_charge(fighter: &mut L2CFighterCommon, motion_kind: u64) {
+    if fighter.is_motion_one_of(&[
+        Hash40::new("special_lw_start"),
+        Hash40::new("special_air_lw_start"),
+        ]) {
+
+            let is_hold = ControlModule::check_button_on(fighter.module_accessor, *CONTROL_PAD_BUTTON_SPECIAL);
+            let charge = VarModule::get_int(fighter.battle_object, vars::dedede::instance::JET_CHARGE_FRAMES) as f32;
+            let max_charge_frames = WorkModule::get_param_float(fighter.module_accessor, hash40("param_special_lw"), hash40("hold_max_f"));
+            let charge_start_frame = 6.0;
+            let charge_end_frame = 11.0;
+
+        if (charge_start_frame..charge_end_frame).contains(&fighter.motion_frame()) && charge < (max_charge_frames) && is_hold{
+            let mut rand = app::sv_math::randf(hash40("fighter"), 1.0);
+            let mut rand_waist = app::sv_math::randf(hash40("fighter"), 1.0);
+
+            if ArticleModule::is_exist(fighter.module_accessor, *FIGHTER_DEDEDE_GENERATE_ARTICLE_JETHAMMER){
+                let article = ArticleModule::get_article(fighter.module_accessor, *FIGHTER_DEDEDE_GENERATE_ARTICLE_JETHAMMER);
+                let object_id = smash::app::lua_bind::Article::get_battle_object_id(article) as u32;
+                let article_boma = sv_battle_object::module_accessor(object_id);
+                if MotionModule::motion_kind(article_boma) != hash40("hold"){
+                    MotionModule::change_motion(article_boma, Hash40::new("hold"), 0.0, 1.0, false, 0.0, false, false);
+                }
+            }
+
+            ModelModule::set_joint_translate(fighter.module_accessor, Hash40::new("shoulderr"), &Vector3f{x:3.0, y:rand, z:-1.0}, false, false); 
+            ModelModule::set_joint_translate(fighter.module_accessor, Hash40::new("waist"), &Vector3f{x:3.0, y:0.0, z:rand_waist}, false, false); 
```

Comment:
random programmatic animation adjustments in opff... I hate it but it might be necessary for what you're doing? Honestly DSpecial charging should work based on two separate animations and statuses like a smash attack (but I won't ask you to make that change)